src/main/java/hudson/plugins/spotinst/cloud/BaseSpotinstCloud.java

```diff
@@ -643,6 +732,179 @@ protected Integer getPendingThreshold() {
+    //                String redisResponseValue = (String) redisGetValueResponse.getValue();
+    //
+    //                if (redisResponseValue != null) {
+    //                    boolean isGroupBelongToController = redisResponseValue.equals(controllerIdentifier);
+    //
+    //                    if (isGroupBelongToController) {
+    //                        handleGroupManagedByThisController(cloud, controllerIdentifier);
+    //                    }
+    //                    else {
+    //                        LOGGER.info(String.format("group %s does not belong to controller with identifier %s", groupId,
+    //                                                  controllerIdentifier));
+    //                        boolean isContainsCandidates =
+    //                                SpotinstContext.getInstance().getCandidateGroupsForControllerOwnership()
+    //                                               .containsKey(groupId);
+    //
+    //                        if (isContainsCandidates == false) {
+    //                            SpotinstCloudCommunicationState cloudState =
+    //                                    SpotinstContext.getInstance().getCloudsInitializationState().get(cloud);
+    //
+    //                            if (cloudState == null) {
+    //                                SpotinstContext.getInstance().getCandidateGroupsForControllerOwnership()
+    //                                               .put(groupId, accountId);
+    //                                SpotinstContext.getInstance().getCloudsInitializationState()
+    //                                               .put(cloud, SPOTINST_CLOUD_COMMUNICATION_INITIALIZING);
+    //                            }
+    //                        }
+    //                    }
+    //                }
+    //                else {
+    //                    LOGGER.warn("redis response value return null");
+    //                }
+    //            }
+    //            //there is no controller for the given group in redis, should take ownership
+    //            else {
+    //                handleGroupManagedByThisController(cloud, controllerIdentifier);
+    //            }
+    //        }
+    //    }
+
+    public void syncGroupsOwner(BaseSpotinstCloud cloud) {
+        String groupId   = cloud.getGroupId();
+        String accountId = cloud.getAccountId();
+        LOGGER.info(String.format("try fetching controller identifier for group %s from redis", groupId));
+
+        IRedisRepo          redisRepo             = RepoManager.getInstance().getRedisRepo();
+        ApiResponse<Object> redisGetValueResponse = redisRepo.getValue(groupId, accountId);
+        String              controllerIdentifier  = SpotinstContext.getInstance().getControllerIdentifier();
+
+        if (redisGetValueResponse.isRequestSucceed()) {
+            //redis response might return in different types
+            if (redisGetValueResponse.getValue() instanceof String) {
```

Comment:
need to change null, isGroupLocked = redisGetValueResponse.getValue() == null

Reply:
done